### ISSUE TITLE
PHP 7.2 compatibility

### ISSUE THREAD

**Issue by janmuennich** [dotplex/roundcube_calendar#2]
Currently this plugin works with max. PHP 7.1.

/calendar/lib/encryption.php uses mcrypt, which was deprecated in PHP 7.1 and removed in PHP 7.2. The functionality would probably need to be converted to OpenSSL. Don't know if it's possible to use same ciphers etc. so the stored passwords could still be decrypted.

Maybe there are some other issues with PHP 7.2, haven't checked.

**Comment by pafcioooo** [comment]
Well after searching for "mcrypt for php 7.2" there is possibility to install mcrypt in php>=7.2 environment. So maybe this is the way. I have not checked this yet. Once I try I will write here.

**Comment by janmuennich** [comment]
Yes, you can install mcrypt for PHP 7.2. But it is not an official module anymore, so PHP 7.2 compatible scripts should not use mcrypt.

**Comment by pafcioooo** [comment]
@janmuennich it has been 3 months after your initial report, have you been able to some how solve this, then?

Have you seen/tried this post https://github.com/texxasrulez/Caldav_Calendar/issues/17 ?

**Comment by janmuennich** [comment]
I haven't found time for this yet unfortunately, probably won't happen as long as 7.1 is supported.

**Comment by pafcioooo** [comment]
> Have you seen/tried this post texxasrulez/Caldav_Calendar#17 ?

I have manually added entries mentioned in above link into calendar/drivers/caldav/caldav_driver.php and calendar is displayed in RC as well as email with invites are also displaying.
Though there is following error in logs/error.log
```
[17-Mar-2019 09:55:33 +0100]: <e7a7ud1v> DAV Error: Not authenticated in /home/poczta/public_html/roundcubemail/plugins/calendar/lib/SabreDAV/lib/OldSabre/DAV/Client.php on line 425 (POST /?_task=calendar&_action=refresh)
[17-Mar-2019 09:55:33 +0100]: <e7a7ud1v> DAV Error: Not authenticated in /home/poczta/public_html/roundcubemail/plugins/calendar/lib/SabreDAV/lib/OldSabre/DAV/Client.php on line 425 (POST /?_task=calendar&_action=refresh)
```

**Comment by pafcioooo** [comment]
> [17-Mar-2019 09:55:33 +0100]: <e7a7ud1v> DAV Error: Not authenticated in /home/poczta/public_html/roundcubemail/plugins/calendar/lib/SabreDAV/lib/OldSabre/DAV/Client.php on line 425 (POST /?_task=calendar&_action=refresh)
[17-Mar-2019 09:55:33 +0100]: <e7a7ud1v> DAV Error: Not authenticated in /home/poczta/public_html/roundcubemail/plugins/calendar/lib/SabreDAV/lib/OldSabre/DAV/Client.php on line 425 (POST /?_task=calendar&_action=refresh)

It requires just to add again password in gui.
